Dockerfile

```diff
@@ -1,12 +1,12 @@
-FROM golang:alpine AS build
+FROM golang:1.22-alpine AS build
```

Comment:
nit: is it possible to avoid locking to a version? could be nice to have it follow along with updated versions if possible...

Reply:
> could be nice to have it follow along with updated versions if possible

Unfortunately, it won't. Since builds only happen when we update this software, leaving this unlocked means we'll only notice that we need to do some really unfortunate changes at a time when we're just trying to patch up a bug or add a new feature.

For that reason I prefer to lock the versions (minor versions), especially for runtimes/toolchains/compilers. For these components minor versions are nearly as impactful as major versions, e.g. Go introduced an entirely new way to create modules back in v1.13 that made us have to suddenly change how tilenol manages packages and gets built (see #33) and later on in v1.18 they introduced syntax to support type generics which broke some dependencies (see #52).

In fact, this is why I added the version in this commit: the image wouldn't build against the latest version of Go (v1.23 came out last month) since it introduced a new wave of module and toolchain changes.

> maybe same question for ./start_local_es.sh

As for Elasticsearch, there are a couple of issues here:

1. Elasticsearch major versions (and sometimes minor versions, though to a lesser degree) are not necessarily indicative of client support. Elasticsearch has both cluster software target versions and "wire compatibility" versions. Technically, tilenol only cares about the latter, but client libraries are even a little more specific about this.
2. Elasticsearch does not publish any "latest" version tag or even "8"/"8.15"/etc. This is likely because it's generally not a good idea for the cluster software to automatically upgrade.